lib/processing.py

```diff
@@ -0,0 +1,490 @@
+                        "fuzzy_matching_lastname": lastname_similarity * 100}
+            else:
+                highest_name_simitality, highest_lastname_simitality, highest_name, highest_lastname, extracted_text, bounding_boxes, name_similarity, lastname_similarity = modify_angle_extract_data(config, file_path, name, lastname, highest_name_simitality, highest_lastname_simitality)
+                if name_similarity > 0.9 and lastname_similarity > 0.9:
+                    return {"Name_extracted": highest_name, 
+                            "last_name_extracted": highest_lastname, 
+                            "Bounding_boxes_name": get_matching_bounding_boxes(highest_name, bounding_boxes), 
+                            "Bounding_boxes_last_name": get_matching_bounding_boxes(highest_lastname, bounding_boxes),
+                            "fuzzy_matching_name": name_similarity * 100,
+                            "fuzzy_matching_lastname": lastname_similarity * 100}
+                else:
+                    return {"outcome": f"the highest name similarity is: {highest_name_simitality} with the extract name {highest_name}  and the highest lastname similarity is: {highest_lastname_simitality} with the extract lastname {highest_lastname}"}
+
+
+## FUNTIONS TO CREATE A RAG 
+def normalize_text(s):
+    """
+    Normalize the text by removing special characters and multiple spaces.
+    
+    Args:
+        s (str): The input text.
+    
+    Returns:
+        str: The normalized text.
+    """
+    s = re.sub(r'\s+', ' ', s).strip()
+    s = re.sub(r". ,", "", s)
+    s = s.replace("..", ".")
+    s = s.replace(". .", ".")
+    s = s.replace("\n", "")
+    s = s.replace("\t", "")
+    s = s.replace("  ", " ")
+    s = s.replace("_", "")
+    s = s.replace("#", "")
+    s = s.strip()
+    
+    return s
+
+# Create an generate embeddings
+def create_embeddings(client, text, tokenizer, model):
+    """
+    Create embeddings for a given document file.
+    
+    Args:
+        client (OpenAI): The OpenAI client.
+        file_path (str): The path to the document file.
+    
+    Returns:
+        tuple: A tuple containing the embedding, tokens, and normalized text. 
+               Returns a string when the file format is invalid.
+    """
+    if len(text) > 0:
+        
+        # Normalize the text
+        normalized_text = normalize_text(text)
+        # Tokenize the normalized text
+        tokenizer = tiktoken.get_encoding(tokenizer)
+        tokens = tokenizer.encode(normalized_text)
+
+        text = list(normalized_text.split(" "))
+        embeddings_list = []
+        chunk_text = []
+        #Create embeddings using OpenAI client
+        if len(tokens) > 500:
+            total_chunks = len(tokens) // 500
+            # text = list(normalized_text.split(" "))
+            for i in range(total_chunks):
+                embedding = client.embeddings.create(input=' '.join(text[i*500:(i+1)*500]), model=model).data[0].embedding
```

Comment:
There is ChatOpenAI and EmbeddingsOpenAI you could have use that one and avoid the *novo* implementation